resources/abi/README.md

```diff
@@ -0,0 +1,74 @@
+# ABI change monitoring
+
+When the OpenAssetIO C++ ABI changes, we need to know about it to inform
+our release notes and semver versioning.
+
+This is aided (though not solved!) by automated testing against a
+snapshot dump of the ABI. The tool we use for this is [libabigail](https://sourceware.org/libabigail/manual/libabigail-overview.html),
+in particular its `abidiff` and `abidw` tools.
+
+> **Note**
+>
+> The ABI on other platforms, including other Linux distributions, may
+> (subtly) vary. Always generate a dump of a build from the same Docker
+> image as is used on CI, i.e. `ghcr.io/openassetio/openassetio-build`.
+
+If the ABI of an OpenAssetIO library changes, and the change is
+unavoidable, then a new XML snapshot dump of the ABI must be created
+(and release notes updated as appropriate).
+
+To do this we use `abidw`. For convenience, the `openassetio-build`
+Docker image has libabigail pre-installed.
+
+The generated snapshot is then checked against new builds using
+`abidiff`.
+
+## Example of updating the ABI snapshot
+
+Assuming:
+* The current working directory is at the root of the repository.
+* A POSIX host (for Docker invocation).
+* The library with an ABI break is `libopenassetio`.
+
+First we create a new build using the `openassetio-build` Docker image
+
+```sh
+docker run --rm -v $(pwd):/src \
+    ghcr.io/openassetio/openassetio-build \
+    bash -c 'cd /src && \
+        cmake -S . -B build -DCMAKE_BUILD_TYPE=Debug && \
+        cmake --build build --parallel'
+```
+
+From this we can generate a new ABI dump
+
+```sh
+docker run --rm -v $(pwd):/src \
+    ghcr.io/openassetio/openassetio-build \
+    abidw /src/build/lib64/libopenassetio.so.1.0.0 \
+    > resources/abi/libopenassetio.so.1.0.0.xml
+```
+
+This will generate a new XML formatted ABI dump to overwrite the
+existing XML dump under `resources/abi`.
+
+We can then check against this using `abidiff`
+
+```sh
+docker run --rm -v $(pwd):/src \
+    ghcr.io/openassetio/openassetio-build \
+    abidiff /src/resources/abi/libopenassetio.so.1.0.0.xml \
+    /src/build/lib64/libopenassetio.so.1.0.0
+```
+
+or re-run the ABI tests via CTest
+
+```sh
+docker exec openassetio-build \
+    bash -c 'cd /src && \
+        cmake -S . -B build -DCMAKE_BUILD_TYPE=Debug \
+          -DOPENASSETIO_ENABLE_TESTS=ON \
+          -DOPENASSETIO_ENABLE_TEST_ABI=ON && \
+        ctest --test-dir build --output-on-failure \
+          -R openassetio.test.abi'
+```
```

Comment:
[Consider] Missing a step to say the dump should be commited back to the git repo if an ABI break is expected and desired?